code/game/objects/items/surgery_tray.dm

```diff
@@ -0,0 +1,207 @@
+				continue
+			// the existing tool was worse than us, ditch it
+			surgery_overlays -= surgery_tool_type
+			LAZYSET(recorded_tool_speeds, surgery_tool_type, surgery_tool.toolspeed)
+			surgery_overlays[surgery_tool_type] = actual_overlay
+
+	for(var/surgery_tool in surgery_overlays)
+		. |= surgery_overlays[surgery_tool]
+
+///Spawn the things we contain on initialisation
+/obj/item/surgery_tray/proc/populate_contents()
+	for (var/thing_path in initial_contents)
+		new thing_path(src)
+	update_appearance(UPDATE_OVERLAYS)
+
+///Sets the surgery tray's deployment state. Silent if user is null.
+/obj/item/surgery_tray/proc/set_tray_mode(new_mode, mob/user)
+	is_portable = new_mode
+	density = !is_portable
+	if(user)
+		user.visible_message(span_notice("[user] [is_portable ? "retracts" : "extends"] [src]'s wheels."), span_notice("You [is_portable ? "retract" : "extend"] [src]'s wheels."))
+
+	if(is_portable)
+		interaction_flags_item |= INTERACT_ITEM_ATTACK_HAND_PICKUP
+		pass_flags |= PASSTABLE
+		RemoveElement(/datum/element/noisy_movement)
+	else
+		interaction_flags_item &= ~INTERACT_ITEM_ATTACK_HAND_PICKUP
+		pass_flags &= ~PASSTABLE
+		AddElement(/datum/element/noisy_movement)
+
+	update_appearance()
+
+/obj/item/surgery_tray/equipped(mob/user, slot, initial)
+	. = ..()
+	if(!is_portable)
+		set_tray_mode(TRUE, user)
+
+/obj/item/surgery_tray/attack_self(mob/user, modifiers)
+	. = ..()
+	if(.)
+		return
+	var/turf/open/placement_turf = get_turf(user)
+	if(isgroundlessturf(placement_turf) || isclosedturf(placement_turf))
+		balloon_alert(user, "can't deploy!")
+		return TRUE
+	if(!user.transferItemToLoc(src, placement_turf))
+		balloon_alert(user, "tray stuck!")
+		return TRUE
+	set_tray_mode(FALSE, user)
+	return
+
+/obj/item/surgery_tray/attack_hand(mob/living/user)
+	if(!user.can_perform_action(src, NEED_HANDS))
+		return ..()
+	var/obj/item/grabbies = pick(contents)
+	if(grabbies)
+		atom_storage.remove_single(user, grabbies, drop_location())
+		user.put_in_hands(grabbies)
+	return TRUE
+
+/obj/item/surgery_tray/dump_contents()
+	var/atom/drop_point = drop_location()
+	for(var/atom/movable/tool in contents)
+		tool.forceMove(drop_point)
+
+/obj/item/surgery_tray/deconstruct(disassembled = TRUE)
+	dump_contents()
+	return ..()
+
+/obj/item/surgery_tray/morgue
+	name = "autopsy tray"
+	desc = "A Deforest brand surgery tray, made for use in morgues. It is a folding model, \
+		meaning the wheels on the bottom can be extended outwards, making it a cart."
+	initial_contents = list(
+		/obj/item/blood_filter,
+		/obj/item/bonesetter,
+		/obj/item/cautery/cruel,
+		/obj/item/circular_saw,
+		/obj/item/clothing/mask/surgical,
+		/obj/item/hemostat/cruel,
+		/obj/item/razor/surgery,
+		/obj/item/retractor/cruel,
+		/obj/item/scalpel/cruel,
+		/obj/item/stack/medical/bone_gel,
+		/obj/item/stack/sticky_tape/surgical,
+		/obj/item/surgical_drapes,
+		/obj/item/surgicaldrill,
+	)
+
+/datum/storage/medicart
```

Comment:
I believer this should be in the `datum/storage/subtypes` folder but whatever...

Reply:
should the New() also be moved to that file?